cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -473,10 +473,16 @@ public RunRecordMeta recordProgramDeprovisioning(ProgramRunId programRunId, byte
     ProgramRunClusterStatus clusterStatus = existing.getCluster().getStatus();
     delete(existing);
     key.add(getInvertedTsKeyPart(existing.getStartTs())).add(programRunId.getRun()).build();
+    ProgramRunStatus status = existing.getStatus();
 
-    // if the previous state was provisioning, that means we've transitioned here from a failure
-    ProgramRunStatus newStatus = clusterStatus == ProgramRunClusterStatus.PROVISIONING ?
-      ProgramRunStatus.FAILED : existing.getStatus();
+    // if the previous state was provisioning, that means we've transitioned here from a failure or from a stop
+    // if we've transitioned here from a stop, the program run status will be killed.
```

Comment:
Since we are transitioning the program to failed here, it would be good to say where the killed transition is done. Also, this may be something for the future - why are these transitions not in the same place?